api/envoy/api/v2/route/route.proto

```diff
@@ -377,34 +377,49 @@ message RouteAction {
-  //   :ref:`route's match <envoy_api_field_route.Route.match>` prefix value.
-  //   Stripping a prefix from a path requires multiple Routes to handle all cases. For example,
-  //   rewriting */prefix* to */* and */prefix/etc* to */etc* cannot be done in a single
-  //   :ref:`Route <envoy_api_msg_route.Route>`, as shown by the below config entries:
-  //
-  //   .. code-block:: yaml
-  //
-  //     - match:
-  //         prefix: "/prefix/"
-  //       route:
-  //         prefix_rewrite: "/"
-  //     - match:
-  //         prefix: "/prefix"
-  //       route:
-  //         prefix_rewrite: "/"
-  //
-  //   Having above entries in the config, requests to */prefix* will be stripped to */*, while
-  //   requests to */prefix/etc* will be stripped to */etc*.
-  string prefix_rewrite = 5;
+  oneof path_rewrite_specifier {
+    // Indicates that during forwarding, the matched prefix (or path) should be
+    // swapped with this value. This option allows application URLs to be rooted
+    // at a different path from those exposed at the reverse proxy layer. The router filter will
+    // place the original path before rewrite into the :ref:`x-envoy-original-path
+    // <config_http_filters_router_x-envoy-original-path>` header.
+    //
+    // .. attention::
+    //
+    //   Pay careful attention to the use of trailing slashes in the
+    //   :ref:`route's match <envoy_api_field_route.Route.match>` prefix value.
+    //   Stripping a prefix from a path requires multiple Routes to handle all cases. For example,
+    //   rewriting */prefix* to */* and */prefix/etc* to */etc* cannot be done in a single
+    //   :ref:`Route <envoy_api_msg_route.Route>`, as shown by the below config entries:
+    //
+    //   .. code-block:: yaml
+    //
+    //     - match:
+    //         prefix: "/prefix/"
+    //       route:
+    //         prefix_rewrite: "/"
+    //     - match:
+    //         prefix: "/prefix"
+    //       route:
+    //         prefix_rewrite: "/"
+    //
+    //   Having above entries in the config, requests to */prefix* will be stripped to */*, while
+    //   requests to */prefix/etc* will be stripped to */etc*.
+    string prefix_rewrite = 5;
+
+    // [#not-implemented-hide:]
+    // Indicates that during forwarding, paths that match the pattern should be
+    // rewritten, interpolating match groups from the pattern into the new path as specified.
+    // This option allows application URLs to be rewritten at reverse proxy layer in a way that is
+    // aware of segments with variable contents like IDs.
+    // The router filter will place the original path before rewrite into the
+    // :ref:`x-envoy-original-path // <config_http_filters_router_x-envoy-original-path>` header.
+    // Examples:
+    //
+    // * The pattern and subsitution string *^/foo/(bar)/(.*)$ /$1/$2* would transform
+    // */foor/bar/baz/bang* into */bar/baz/bang*.
+    string path_rewrite_regex = 25;
```

Comment:
I think it would be better to make the type of this field a message containing two strings, rather than depend on a specific string format.

Reply:
+1
message regex_rewrite
{
  string pattern
  string substitution
}